### PR TITLE
Personalize Message on Subject Dashboard, Saving Quiz Data to DB

### PR DESCRIPTION
**Pull request by aaryanptl**

### CODE REVIEW

File: src/actions/language.actions.ts [hunk 1]
```diff
@@ -14,10 +18,24 @@ export const getLanguageLevels = async () => {
...
-  const { data, error } = await supabase.from("languages_topics").select("*");
+  const { data, error } = await supabase
+    .from("languages_topics")
+    .select("*, languages_db(*), languages_quiz(*)")
+    .eq("languages_db.language_id", langId)
+    .eq("languages_quiz.user_id", userId)
+    .eq("languages_quiz.language_id", langId)
+    .filter("languages_db", "not.is", null);
+
+  console.log(data);
```
Comment (raghav-dwivedi) [inline src/actions/language.actions.ts]: remove console

File: src/actions/language.actions.ts [hunk 2]
```diff
@@ -14,10 +18,24 @@ export const getLanguageLevels = async () => {
...
+}) => {
   const supabase = createClient();
 
-  const { data, error } = await supabase.from("languages_topics").select("*");
+  const { data, error } = await supabase
+    .from("languages_topics")
+    .select("*, languages_db(*), languages_quiz(*)")
+    .eq("languages_db.language_id", langId)
+    .eq("languages_quiz.user_id", userId)
+    .eq("languages_quiz.language_id", langId)
```
Comment (raghav-dwivedi) [inline src/actions/language.actions.ts:35]: checking language_id twice

File: src/actions/language.actions.ts [hunk 3]
```diff
@@ -56,10 +78,292 @@ export const getTopicContent = async ({
     .from("languages_db")
     .select("*")
     .eq("topic_id", topic)
-    .eq("language_id", languageId?.id);
+    .eq("language_id", languageId?.id)
+    .order("id", { ascending: true })
+    .range(from || 0, to || 20);
+
+  if (error) {
+    console.log(error);
```
Comment (raghav-dwivedi) [inline src/actions/language.actions.ts]: remove console

File: src/actions/language.actions.ts [hunk 4]
```diff
@@ -56,10 +78,292 @@ export const getTopicContent = async ({
...
+  quizId: number;
+  state: number;
+}) => {
+  const supabase = createClient();
+
+  console.log({
+    quizId,
+    userId,
+    total,
+    correct,
```
Comment (SachinShetty11493) [inline src/actions/language.actions.ts]: Remove console

File: src/app/(dashboard)/(routes)/languages/components/language-dashboard.tsx [hunk 5]
```diff
@@ -29,36 +22,83 @@ type Props = {
...
   const [avatar, setAvatar] = useState<string>("");
-  const [topicLoader, setTopicLoader] = useState<boolean>(false);
-  const [dashboardLoader, setDashboardLoader] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (activityData?.response) {
+      if (activityData.response.currentStudentMeta?.pic) {
+        setAvatar(activityData.response.currentStudentMeta.pic);
+        localStorage.setItem(
+          "user-avatar",
```
Comment (SachinShetty11493) [inline src/app/(dashboard)/(routes)/languages/components/language-dashboard.tsx:59]: This is set but not used anywhere, why is this required

File: src/app/(dashboard)/(routes)/languages/learn/page.tsx [hunk 6]
```diff
@@ -1,83 +1,61 @@
...
+      topic: searchParams.topic,
+      language: searchParams.lang,
+      from,
+      to,
+    });
+
+    if (!content) {
+      redirect("/languages");
+    }
+
```
Comment (SachinShetty11493) [inline src/app/(dashboard)/(routes)/languages/learn/page.tsx:45]: Do we need loader here?

**Review by SachinShetty11493 — CHANGES_REQUESTED** [review]

File: src/actions/language.actions.ts (hunk 2 above)
Comment (aaryanptl) [reply src/actions/language.actions.ts:35]: these are two different tables, languages_db and languages_quiz.

File: src/app/(dashboard)/(routes)/languages/components/language-dashboard.tsx (hunk 5 above)
Comment (aaryanptl) [reply src/app/(dashboard)/(routes)/languages/components/language-dashboard.tsx:59]: avatar is being used on quiz pages, and I didn't add it
